### PR TITLE
added exit on close option

### PR DESCRIPTION
**Pull request by dominguesgm**
Added advanced option to quit tomahawk on window close. Option "Exit on close" defaults to false and can be changed by the user.

### CODE REVIEW

File: src/tomahawk/TomahawkWindow.cpp [hunk 1]
```diff
@@ -769,7 +781,7 @@ TomahawkWindow::closeEvent( QCloseEvent* e )
     }
 #endif
 
-    QMainWindow::closeEvent( e );
+    QMainWindow::closeEvent(e);
```
Comment (dschmidt) [inline src/tomahawk/TomahawkWindow.cpp]: The old version is the style we use throughout Tomahawk, so please change this back.

File: src/tomahawk/TomahawkWindow.cpp [hunk 2]
```diff
@@ -757,11 +757,23 @@ TomahawkWindow::changeEvent( QEvent* e )
...
+
+	if(s->exitOnClose()){
+        qApp->quit();
+        return;
+    }
+
 #ifndef Q_OS_MAC
-    if ( e->spontaneous() && QSystemTrayIcon::isSystemTrayAvailable() )
+    // CHANGED
+    if (!s->exitOnClose() && e->spontaneous() && QSystemTrayIcon::isSystemTrayAvailable())
```
Comment (dschmidt) [inline src/tomahawk/TomahawkWindow.cpp]: a) you removed whitespaces that we like.
b) is this even neccessary? If the diff view in Github isn't fooling me, you're `return`ing anyway if `exitOnClose()` is true, so this code is never able to run. To make it more obvious you could make it `else if()` in the `Q_OS_MAC` section.

File: src/tomahawk/TomahawkWindow.cpp [hunk 3]
```diff
@@ -757,11 +757,23 @@ TomahawkWindow::changeEvent( QEvent* e )
...
 
 
+// CHANGED
 void
 TomahawkWindow::closeEvent( QCloseEvent* e )
 {
+
+    // CHANGED change according to definitions
+    TomahawkSettings* s = TomahawkSettings::instance();
+    //QMainWindow::setQuitOnLastWindowClosed(s->exitOnClose());
```
Comment (dschmidt) [inline src/tomahawk/TomahawkWindow.cpp]: Why is this code commented out? If it doesn't work don't add it at all, or at least leave a comment why it's commented out

**Comment by dschmidt** [comment]
Why are you adding all these `//CHANGED` comments?

**Comment by DominguesGM** [comment]
Ok, some changes that were missing were fixed and I changed the style of the code as you said.

File: src/tomahawk/TomahawkWindow.cpp [hunk 4]
```diff
@@ -760,16 +760,26 @@ TomahawkWindow::changeEvent( QEvent* e )
...
+
+#ifndef Q_OS_MAC
+    else {
+        if ( e->spontaneous() && QSystemTrayIcon::isSystemTrayAvailable() )
+        {
+            hide();
+            e->ignore();
+            return;
+        }
+    }
```
Comment (dschmidt) [inline src/tomahawk/TomahawkWindow.cpp]: No need to use an if in an else block here, just use else if

File: src/tomahawk/TomahawkWindow.cpp [hunk 5]
```diff
@@ -760,16 +760,26 @@ TomahawkWindow::changeEvent( QEvent* e )
...
 #endif
 
-    QMainWindow::closeEvent( e );
+    QMainWindow::closeEvent( e );    
 }
```
Comment (dschmidt) [inline src/tomahawk/TomahawkWindow.cpp]: Trailing whitespaces

**Comment by dschmidt** [comment]
Cool. Almost perfect now!

File: src/tomahawk/TomahawkWindow.cpp (hunk 4 above)
Comment (DominguesGM) [inline src/tomahawk/TomahawkWindow.cpp]: So just removing the outer brackets?

Comment (dschmidt) [inline src/tomahawk/TomahawkWindow.cpp]: Yeah

**Comment by dschmidt** [comment]
And when you're done, please squash all commits into one

**Comment by dschmidt** [comment]
Can you squash your commits into one? Then I will merge.

**Comment by dschmidt** [comment]
ping @DominguesGM

**Comment by muesli** [comment]
Yeah, looks good. @DominguesGM wanna do the commit squashing or need help with it?

**Comment by hugolm84** [comment]
@dominguesgm  Hi! If you don't mind, please squash and then I'll merge. :)

**Comment by dschmidt** [comment]
Well, github has (maybe by now) a squash and merge button, we can use that and finally land this after little less than 5 years :)

**Pull request merged**